orders/main.go

```diff
@@ -0,0 +1,163 @@
+	defer httpCancel()
+	InfoLogger.Print("shutting down the gin HTTP server")
+	if err := h.Shutdown(httpCtx); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	InfoLogger.Print("shutting down the NATS connection")
+	if err := n.Close(); err != nil {
+		errs = append(errs, err.Error())
+	}
+
+	mongoCtx, mongoCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer mongoCancel()
+	InfoLogger.Print("shutting down the MongoDB connection")
+	if err := mongoClient.Disconnect(mongoCtx); err != nil {
+		errs = append(errs, err.Error())
+	}
+	return errs
+}
+
+type mainConfig map[string]string
+
+func genMainConfig() (mainConfig, []string) {
+	var missingEnvs []string
+	conf := mainConfig{}
+	envToErrString := map[string]string{
+		"MONGO_CONN_STR":  "missing mongo connection: MONGO_CONN_STR",
+		"JWT_SIGN_KEY":    "missing JWT HS256 signing key: JWT_SIGN_KEY",
+		"NATS_CLUSTER_ID": "missing NATS cluster ID: NATS_CLUSTER_ID",
+		"NATS_CLIENT_ID":  "missing NATS client ID: NATS_CLIENT_ID",
+		"NATS_CONN_STR":   "missing NATS connection string: NATS_CONN_STR",
+	}
+	for key, errStr := range envToErrString {
+		if val, ok := os.LookupEnv(key); !ok {
+			missingEnvs = append(missingEnvs, errStr)
+		} else {
+			conf[key] = val
+		}
+	}
+	return conf, missingEnvs
+}
+
+func main() {
+	// parse environment variables for startup info
+	conf, missingEnvs := genMainConfig()
+	if len(missingEnvs) > 0 {
+		for _, errStr := range missingEnvs {
+			ErrorLogger.Print(errStr)
+		}
+		os.Exit(1)
+	}
+
+	// init MongoDB collections
+	ctx, cancel := context.WithTimeout(context.Background(), dbTimeout)
+	defer cancel()
+	client, err := mongo.NewClient(options.Client().ApplyURI(conf["MONGO_CONN_STR"]))
+	if err != nil {
+		ErrorLogger.Printf("mongo.NewClient: %v", err)
+		os.Exit(1)
+	}
+	if err := client.Connect(ctx); err != nil {
+		ErrorLogger.Printf("mongo client.Connect: %v", err)
+		os.Exit(1)
+	}
+	if err := client.Ping(ctx, nil); err != nil {
+		ErrorLogger.Printf("mongo client.Ping: %v", err)
+		os.Exit(1)
+	}
+	InfoLogger.Print("connected to MongoDB")
+	db := client.Database(dbName)
+
+	tc := newTicketCollection(db.Collection(ticketCollectionName), dbTimeout)
+	oc := newOrdersCollection(db.Collection(ordersCollectionName), dbTimeout)
+
+	// init NATS Streaming Server connection
+	natsClient, err := stan.Connect(conf["NATS_CLUSTER_ID"], conf["NATS_CLIENT_ID"], stan.NatsURL(conf["NATS_CONN_STR"]))
+	if err != nil {
+		ErrorLogger.Printf("unable to connect to NATS Streaming Server: %v", err)
+		os.Exit(1)
+	}
+	InfoLogger.Print("able to connect to NATS Streaming Server")
+	defer func() {
+		if err := natsClient.Close(); err != nil {
+			panic(err)
+		}
+	}()
+
+	// create gin router and bind handlers/routes to it
+	gin.SetMode(gin.ReleaseMode)
+	r := gin.Default()
+	server, err := newApiServer(conf["JWT_SIGN_KEY"], 15*time.Minute, r, tc, oc, natsClient)
+	if err != nil {
```

Comment:
Mongo and NATS connections exist at this point, so those should be gracefully closed before exiting